packages/terra-functional-testing/lib/services/wdio-docker-service.js

```diff
@@ -0,0 +1,167 @@
+/* eslint-disable class-methods-use-this */
+const path = require('path');
+const util = require('util');
+const childProcess = require('child_process');
+const Logger = require('../logger/logger');
+
+const exec = util.promisify(childProcess.exec);
+const logger = new Logger({ prefix: 'wdio-docker-service' });
+
+const RETRY_COUNT = 30;
+const POLL_INTERVAL = 2000;
+
+class DockerService {
+  /**
+   * Prepares the docker testing environment.
+   */
+  async onPrepare(config) {
+    this.host = config.hostname;
+    this.port = config.port;
+
+    await this.initializeSwarm();
+    await this.deployStack();
+  }
+
+  /**
+   * Initializes a docker swarm instance.
+   */
+  async initializeSwarm() {
+    const { stdout: dockerInfo } = await exec('docker info --format "{{json .}}"');
+    const { Swarm } = JSON.parse(dockerInfo);
+
+    if (Swarm.LocalNodeState !== 'active') {
+      logger.log('Initializing docker swarm.');
+
+      await exec('docker swarm init');
+    }
+  }
+
+  /**
+   * Deploys the docker stack.
+   */
+  async deployStack() {
+    // Remove the previous stack if one exists.
+    await this.removeStack();
+
+    logger.log('Deploying docker stack.');
+
+    const composeFilePath = path.resolve(__dirname, '../docker/docker-compose.yml');
+
+    await exec(`docker stack deploy -c ${composeFilePath} wdio`);
+
+    await this.awaitNetworkReady();
+  }
+
+  /**
+   * Removes the docker stack.
+   */
+  async removeStack() {
+    const { stdout: stackInfo } = await exec('docker stack ls | grep wdio || true');
+
+    if (stackInfo) {
+      logger.log('Removing docker stack.');
+
+      await exec('docker stack rm wdio');
+
+      // Ensure the services and network have been removed.
+      await this.awaitServiceRemoval();
+      await this.awaitNetworkRemoval();
+    }
+  }
+
+  /**
+   * Waits for a command to complete successfully.
+   * @param {string} command - The shell command to run.
+   * @param {func} callback - A callback function to accept or reject the result of the command. Must return a promise.
+   * @param {number} retries - The number of times to retry the command. Defaults to 30 retries.
+   * @param {number} interval - The timeout between commands in milliseconds. Defaults to every two seconds.
+   * @returns {Promise} - A promise that resolves when the callback accepts the command response.
+   */
+  async pollCommand(command, callback, retries = RETRY_COUNT, interval = POLL_INTERVAL) {
```

Comment:
Why not like the following?
```suggestion
  async pollCommand(command, callback, retries = 30, interval = 2000) {
```

Reply:
Originally these weren't used in the parameters and I wanted to avoid magic numbers. I've removed the constants and updated these to match your parameter suggestions:

https://github.com/cerner/terra-toolkit/pull/412/commits/81ec6b8b522dd42261437c12f789f6e373615c9a